### ISSUE TITLE
Convention for Missing Clinical Event Dates

### ISSUE THREAD

**Issue by katy-sadowski** [OHDSI/Themis#82]
**Issue Name:** 
We're missing an official convention for how to handle clinical events that are missing a date in the source data.  Since the x_start_date field is required in all clinical event tables, we should provide guidance for users to ensure missing dates are handled consistently across CDMs. In this issue the focus will be on the following tables:

- condition_occurrence
- drug_exposure
- procedure_occurrence
- device_exposure
- observation

These types of events may be found in medical history / patient reported contexts in which they may not have dates recorded.  For other types of events (visits, measurements) the use case & implementation of such a convention is less clear so we'll focus on these 5 in this issue.

**Issue Type:** Lack of an ETL convention for a CDM table and/or field

**Description:** 
Current documentation:
- https://ohdsi.github.io/CommonDataModel/cdm54.html#CONDITION_OCCURRENCE; https://ohdsi.github.io/CommonDataModel/cdm54.html#DRUG_EXPOSURE; https://ohdsi.github.io/CommonDataModel/cdm54.html#PROCEDURE_OCCURRENCE; https://ohdsi.github.io/CommonDataModel/cdm54.html#DEVICE_EXPOSURE - these docs do not describe how to handle source records w/ missing (start) date
- https://ohdsi.github.io/CommonDataModel/cdm54.html#OBSERVATION - for observation_start_date, states "For some observations the ETL may need to make a choice as to which date to choose."

We should update docs with explicit instructions on how to handle missing dates under certain circumstances.  Given the "junk drawer" nature of OBSERVATION we won't be able to cover all circumstances, but there are some well-understood use cases like "history of" events we could make a convention for.

**Related forum post:** https://forums.ohdsi.org/t/how-to-handle-missing-dates-in-all-cdm-tables/9659

The reason I re-kindled that thread is that in DQD we're trying to provide guidance to users on how to resolve failing data quality checks.  For isRequired I'd like to be able to point to CDM docs in order to recommend how users should handle rows w/ a NULL x_start_date (at least in certain cases).

**Suggested solution:** 
For conditions, drug, procedures, devices missing a date in EHR we could recommend putting them in the observation table as "history of" observations (a common scenario is for historical diagnoses to be recorded on a problem list without a date). We'll need a convention for how to populate the observation start date for these events (perhaps: if they are associated with a visit, use the visit start date, otherwise ???). 

I'm curious if there are other scenarios we can/should address in addition to the one above; this may be worth discussing.

**Reported By:** Katy

**Comment by MelaniePhilofsky** [comment]
@katy-sadowski Do we want to discuss on a Data Quality WG call or a Themis WG call?

**Comment by katy-sadowski** [comment]
Yes!  But personally my plate is too full to push this forward right now...things are about to get crazy with the Symposium, and then the holidays 😅 maybe we can put this on a Themis call in the new year?

**Comment by MelaniePhilofsky** [comment]
Yes, no rush! 2025 it is :)

**Comment by MelaniePhilofsky** [comment]
@katy-sadowski Are you ready to tackle this?

**Comment by katy-sadowski** [comment]
@MelaniePhilofsky let's do it. Is the next step to have a discussion at a THEMIS meeting?

**Comment by MelaniePhilofsky** [comment]
First we need a use case, @katy-sadowski. Why should we bring these data into the CDM? What question are we trying to answer which requires these data with unknown dates?

**Comment by katy-sadowski** [comment]
The use case top of mind for me is for some exclusion criteria in cohort definitions, we don't care when an event happened as long as it was prior to the index date.  So using the date the event was recorded in the system, or the date of the visit during which it was reported, may be acceptable proxies for the actual event date.  Such events could be placed in the Observation table as "history of" records.  If there is no date whatsoever associated with an event, we probably need to exclude it from the CDM because then it's useless in the use case I mention.

I would guess this is probably stuff that many ETL'ers are doing already.  I just think we should make an explicit convention for it to ensure everyone's approaching this the same way :)

**Comment by MelaniePhilofsky** [comment]
Solid use case, @katy-sadowski!

The Observation table is the place for these data, [Per the CDM rules](https://ohdsi.github.io/CommonDataModel/cdm54.html#observation) the OBSERVATION.observation_date = "The date of when the Observation was obtained".

We will need to loop in the Vocab team for this call to give guidance on which concept_ids to use for each 'history of' event (i.e. condition, medication, procedure, etc.). I'll discuss with them and you offline to ensure availability for the discussion.

**Comment by MelaniePhilofsky** [comment]
@katy-sadowski,
I spoke to the vocabulary team and they are available for our May 15th call. Does this date work with your schedule? If yes, give this post a thumbs up and I will put it on the Themis schedule!

**Comment by MelaniePhilofsky** [comment]
This issue is duplicative of issue #194 "history of". Closing this issue.
